projects/plugins/crm/includes/ZeroBSCRM.DAL3.php

```diff
@@ -4273,6 +4274,9 @@ public function addUpdateTagObjLinks($args=array()){
             // mode
             if (gettype($mode) != 'string' || !in_array($mode, array('replace','append','remove'))) return false;
 
+			// Retrieving segment information in order to update them with tag changes later.
+			$segments = $zbs->DAL->segments->getSegments( $owner, 1000, 0, true ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```

Comment:
We don't need to retrieve the segments here. For example, in the `switch` block below, we don't use the $segments list.

The method `compile_segments_from_tagIDs` will be called once, so why not retrieve the segments there? in that method `compile_segments_from_tagIDs` or maybe `compile_segments_on_tags_update`

Reply:
Oh, good point! I should have moved it over when creating the helper function. Moved now.